manifest-build-tools/application/make_debian_packages.py

```diff
@@ -0,0 +1,232 @@
+        version = version_generator.generate_package_version(is_official_release)
+        if version != None:
+            params = {}
+            params['PKG_VERSION'] = version
+            repo_name = os.path.basename(repo_dir)
+            version_file = "{0}.version".format(repo_name)
+            version_path = os.path.join(repo_dir, version_file)
+            common.write_parameters(version_path, params)
+            return True
+    except Exception, e:
+        raise RuntimeError("Failed to generate version file for {0} \ndue to {1}".format(repo_dir, e))
+
+def run_build_scripts(top_level_dir, repos, jobs=1, sudo_creds=None):
+    """
+    Go into the directory provided and run all the building scripts.
+    :param top_level_dir: Top level directory that stores all the
+                          cloned repositories.
+    :param repos: A list of repositories to be build
+    :param jobs: Number of parallel jobs(build debian packages) to run.
+    :param sudo_creds: the environment variable name of sudo credentials.
+                       for example: SUDO_CRED=username:password
+    :return:
+        exit on failures
+        None on success.
+    """
+    try:
+        builder = DebianBuilder(top_level_dir, repos, jobs=jobs, sudo_creds=sudo_creds)
+        builder.blind_build_all()
+        builder.print_summary_report()
+        builder.print_detailed_report()
+        result = builder.get_build_result()
+        if result:
+            print "Debian building is finished successfully."
+        else:
+            #builder.print_detailed_report()
+            print "Error found during debian building. cannot continue."
+            sys.exit(2)
+    except Exception, e:
+        sys.exit(1)
+
+def get_build_repos(directory):
+    """
+    :param directory: Directory that stores all the cloned repositories.
+    :return: a list which contains the name of repositories under the directory
+    """
+    repos = []
+    for filename in os.listdir(directory):
+        repos.append(filename)
+    return repos
+
+def checkout_repos(manifest, builddir, force, git_credential, jobs):
+    try:
+        manifest_actions = ManifestActions(manifest, builddir, force=force, git_credentials=git_credential, jobs=jobs, actions=["checkout", "packagerefs"])
+        manifest_actions.execute_actions()
+    except Exception, e:
+        print "Failed to checkout repositories according to manifest file {0} \ndue to {1}. Exiting now...".format(manifest, e)
+        sys.exit(1)
+
+def build_debian_packages(build_directory, jobs, is_official_release, sudo_creds):
+    """
+    Build debian packages
+    """
+    try:
+        repos = get_build_repos(build_directory)
+        for repo in repos:
+            repo_dir = os.path.join(build_directory, repo)
+            generate_version_file(repo_dir, is_official_release)
+
+        # Build Debian packages of repositories except RackHD
+        repos.remove("RackHD")
```

Comment:
I think most different part of your PR than original OnRack code are here , right ?
original OnRack mature code should be good.
for this new part here, If it's workable and I'm fine with it.